src/shadowcasting.cpp

```diff
@@ -0,0 +1,659 @@
+#include "shadowcasting.h"
+
+#include <list>
+
+#include "enums.h"
+#include "fragment_cloud.h" // IWYU pragma: keep
+#include "line.h"
+
+struct slope {
+    slope( int rise, int run ) {
+        // Ensure run is always positive for the comparison operators
+        this->run = abs( run );
+        if( run < 0 ) {
+            this->rise = -rise;
+        } else {
+            this->rise = rise;
+        }
+    }
+
+    int rise;
```

Comment:
Fun fact, the shadowcasting area is only 60 deep and 60 wide, so rise and run can be 8-bit.
probably int_least8_t or uint_least8_t

Reply:
Makes sense, I had meant to go back and revise that but forgot.

Reply:
We do have negative slopes in some valid cases, so I'll be using `int_least8_t`

Reply:
FWIW, I think `int_least8_t` is probably overkill.  Cataclysm is already not going to work on any system where that isn't the same as `int8_t`.